### DISCUSSION TITLE
How to manage versioning of models across 3 versions.

### DISCUSSION THREAD

**Discussion by rrevo** [microsoft/typespec#3455] [Q&A]
The Typespec file below does not compile

```
import "@typespec/http";
import "@typespec/rest";
import "@typespec/versioning";

using TypeSpec.Versioning;

@versioned(Versions)
@service({
  title: "Widget Service",
})
namespace DemoService;

enum Versions {
  v1,
  v2,
  v3
}

using TypeSpec.Http;
using TypeSpec.Rest;

model Widget {
  color: "red" | "blue";
  @added(Versions.v2) name: string;  // FAIL
}

@added(Versions.v3)
model Widget3 is Widget {
  fancyField: string
}
```

with the failure
'DemoService.Widget3' was added in version 'v3' but contains type 'DemoService.Widget3.name' added in version 'v2'.

I wanted to model v1 followed by v2 followed by v3. However based on how I've declared the models, it looks like v2 and v3 are considered independent.

How can this versioning dependency be resolved? Or is it a bug in Typespec?

Thanks

**Comment by markcowl** [comment]
This looks like a bug.  Transferring.

Note that, you can use 'extends' in this case, but the meaning is not the same.

**Comment by markcowl** [comment]
Converted to this issue: https://github.com/microsoft/typespec/issues/3639
